### DISCUSSION TITLE
Grouping email alerts for conversations

### DISCUSSION THREAD

**Discussion by jaimeiniesta** [papercups-io/papercups#728] [Ideas]
Hi! We've enabled email alerts for conversations, so when a customer sends a message, we get an email.

This is handy, but it can result in a lot of individual emails in our inbox, all with the same subject `A customer has sent you a message`. It would be better if these emails were grouped together so here are some ideas:

**1.- Change the subject to include the conversation ID, or customer email or name**, like:
 
```
A customer (joe@example.com) has sent you a message
```

or

```
New message in conversation 2313h232324
```

**2.- Avoid sending individual emails for each message**, and instead hold them in the server and send them once per minute, including all the messages in the same email. We can get some inspiration on the new [Oban Pro Chunk Worker](https://hexdocs.pm/oban/chunk.html).

What do you think?

**Comment by reichert621** [comment]
hey @jaimeiniesta! 

I think it definitely makes sense to improve the subject line. I like the idea to batch messages to avoid too many individual emails, but I think some users want the emails to show up as quickly as possible (rather than e.g. debouncing by a minute), so I'd have to think about that a little more 🤔

**Reply by jaimeiniesta** [reply]
Sure, this could be configurable, to have it debounce more often.

**Reply by reichert621** [reply]
Ticket for subject line: https://github.com/papercups-io/papercups/issues/772

Will aim to get this done soon!

**Reply by reichert621** [reply]
@jaimeiniesta just wanted to let you know that the subject line improvements are done now!

**Comment by a8t** [comment]
Sounds like a simple Genserver could do the job for the second idea, if it's the way to go. 

I'm wondering, do the notifications get sent even if a user is looking at the chat and responding at that moment? If so, maybe we could cut those down too. Not sure if we're already using Presence for this.

**Reply by jaimeiniesta** [reply]
Cutting down notifications for seen messages is a great idea too. Yes, a Genserver could do the job here.

**Reply by a8t** [reply]
Was trying to remember what article I read about batching with a Genserver and it was @reichert621's article haha! https://papercups.io/blog/genserver
